adoptedStyleSheets.js

```diff
@@ -5,6 +5,51 @@
     return;
   }
 
+  // Can we rely on document.body
+  let polyfillLoaded = false;
+
+  // Polyfill-level reference to the iframe body
+  let frameBody;
+
+   // Style elements that will be attached to the head
+   // that need to be moved to the iframe
+  const deferredStyleSheets = [];
+
+  // Initialize the polyfill — Will be called on the window's load event
+  function initPolyfill() {
+    // Iframe is necessary because to extract the native CSSStyleSheet object
+    // the style element should be connected to the DOM.
+    const iframe = document.createElement('iframe');
+    iframe.hidden = true;
+    document.body.appendChild(iframe);
+
+    frameBody = iframe.contentWindow.document.body;
+
+    // Since we get the sheet from iframe, we need to patch prototype of the
+    // CSSStyleSheet in iframe as well.
+    updatePrototype(iframe.contentWindow.CSSStyleSheet.prototype);
+
+    // Document body will be observed from the very start to catch all added
+    // custom elements
+    createObserver(document.body);
+
+    // Document has loaded
+    polyfillLoaded = true;
+
+    // Move style elements created before document.body
+    // to the iframe along with future styles
+    deferredStyleSheets.forEach(nativeStyleSheet => {
+      frameBody.append(nativeStyleSheet);
+      nativeStyleSheet.disabled = false;
+    });
+
+    // Clear out the deferredStyleSheets array
+    deferredStyleSheets.length = 0;
+  }
+
+  // Proceed with using the iframe to house style elements
+  window.addEventListener('load', initPolyfill);
```

Comment:
`DOMContentLoaded` event is better here. It is enough to have `body` loaded and it does not wait for scripts or styles loaded.